parser/encoding/rpc.go

```diff
@@ -13,6 +13,14 @@ import (
 	schema "encr.dev/proto/encore/parser/schema/v1"
 )
 
+type Lang int
+
+const (
+	_ Lang = iota
+	GO
+	TypeScript
+)
+
```

Comment:
I'm not sure I agree with moving language-specific logic in here; I'd rather we kept `SrcField` as it appears in the Go code, and then let any language generator make any changes it deems are needed.

For instance, if we had a Swift client, that will have a different naming convention to TypeScript and Go - does it mean that the `encoding/rpc.go` file now also needs to understand that as well?

Instead if we keep it `SrcField` as the Go struct field name; then the various client generators can update as needed

Reply:
Agreed with Dom on a solution